cmd/juju/application/deployer/deployer.go

```diff
@@ -697,27 +667,32 @@ var getModelConfig = func(api ModelConfigGetter) (*config.Config, error) {
 	return config.New(config.NoDefaults, attrs)
 }
 
-func (d *factory) validateCharmSeries(seriesName string, imageStream string) error {
+func (d *factory) validateCharmBase(base corebase.Base, imageStream string) error {
```

Comment:
We're moving away from all this validation on the client side, which reduces the argument it may be needed later.

If we're checking only the supported workload operating systems, what aren't we checking with relation to operating system?

Can the places it's used without overlap with the BaseSelector, use the BaseSelector instead? The determineSeriesForLocalCharm method calls it for local charms, which overlaps with the provided example.

Reply:
That is something I can certainly look into, but this PR is getting bloated as it is. I think I'll raise a bug and fix it in a separate PR

Reply:
https://bugs.launchpad.net/juju/+bug/2032639